src/main/java/org/jenkinsci/plugins/casc/ConfigurationAsCode.java

```diff
@@ -136,24 +143,67 @@ public static void init() throws Exception {
     }
 
     /**
-     * Main entry point to start configuration process
+     * Main entry point to start configuration process.
      * @throws ConfiguratorException Configuration error
      */
     public void configure() throws ConfiguratorException {
+        List<String> configParameters = getBundledCasCURIs();
+        if (!configParameters.isEmpty()) {
+            LOGGER.log(Level.FINE, "Located bundled config YAMLs: {0}", configParameters);
+        }
+
         String configParameter = System.getProperty(
                 CASC_JENKINS_CONFIG_PROPERTY,
                 System.getenv(CASC_JENKINS_CONFIG_ENV)
         );
         if (configParameter == null) {
             if (Files.exists(Paths.get(DEFAULT_JENKINS_YAML_PATH))) {
                 configParameter = DEFAULT_JENKINS_YAML_PATH;
-            } else {
-                // No configuration set nor default config file
-                return;
             }
         }
 
-        configure(configParameter);
+        if (configParameter != null) {
+            // Add external config parameter
+            configParameters.add(configParameter);
+        }
+        if (configParameters.isEmpty()) {
+            LOGGER.log(Level.FINE, "No configuration set nor default config file");
+            return;
+        }
+        configure(configParameters);
+    }
+
+    public List<String> getBundledCasCURIs() {
+        final String cascFile = "casc.yml";
+        final String cascDirectory = "casc.yml.d";
```

Comment:
I don't have great suggestion but don't really like `casc.yml.d` as standard directory name. `configuration.d`?

Reply:
Or jenkins.yaml.d to keep it aligned?

Reply:
good point, LGTM